File: elovate_backend/src/main/java/com/tuwien/elovate/dtos/lobby/FinishedGameDto.java
package com.tuwien.elovate.dtos.lobby;

/*-
 * #%L
 * ELOvate
 * %%
 * Copyright (C) 2024 ELOvate GmbH.
 * %%
 * Copyright (C) 2024 ELOvate GmbH. - All Rights Reserved
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Proprietary and confidential
 * #L%
 */

import com.tuwien.elovate.dtos.game.GameDto;
import com.tuwien.elovate.enums.GameEndResult;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
public class FinishedGameDto {
    private GameDto game;
    private GameEndResult gameEndResult;
    private List<PlayerStatsPostGameDto> teamA;
    private List<PlayerStatsPostGameDto> teamB;
    private LocalDateTime finishTime;
    private String deletedLobby;
}

File: elovate_frontend/src/dtos/finishedGameDto.ts
/*-
 * #%L
 * ELOvate
 * %%
 * Copyright (C) 2024 ELOvate GmbH.
 * %%
 * Copyright (C) 2024 ELOvate GmbH. - All Rights Reserved
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Proprietary and confidential
 * #L%
 */
import {GameDto} from "./Game/gameDto";
import {Time} from "@angular/common";
import {UserDto} from "./userDto";

export interface CustomStat {
  statName: string,
  statValue: string
}

export interface PlayerStatsPostGameDto {
  user: UserDto
  ratingBefore: number
  ratingAfter: number
  customStats: CustomStat[]
}

export interface FinishedGameDto {
  game: GameDto,
  gameEndResult: GameEndResult,
  teamA: PlayerStatsPostGameDto[],
  teamB: PlayerStatsPostGameDto[],
  finishTime: Date,
  deletedLobby: string;
}

export enum GameEndResult {
  TEAM_A_WINNER,
  TEAM_B_WINNER,
  DRAW
}
